test/query_test.js

```diff
@@ -1,452 +1,554 @@
+    });
+
+    return Promise.all([p1, p2, p3]);
+  });
+
+  it('exists', function () {
+    return create().then(function (i) {
+      var ref = i.ref;
+      return assertQuery(query.exists(ref), true).then(function() {
+        return client.query(query.delete_expr(ref));
+      }).then(function() {
+        return assertQuery(query.exists(ref), false);
+      });
+    });
+  });
+
+  it('count', function () {
+    var p1 = create({ n: 123 });
+    var p2 = create({ n: 123 });
+    var instances = nSet(123);
     // `count` is currently only approximate. Should be 2.
-    assert.typeOf(await client.query(query.count(instances)), 'number')
-  })
+    return Promise.all([p1, p2]).then(function() {
+      return client.query(query.count(instances)).then(function (count) {
+        assert.typeOf(count, 'number');
+      });
+    });
+  });
 
   // Write functinos
 
-  it('create', async () => {
-    const instance = await create()
-    assert('ref' in instance)
-    assert('ts' in instance)
-    assert.deepEqual(instance.class, classRef)
-  })
-
-  it('update', async () => {
-    const ref = (await create()).ref
-    const got = await client.query(query.update(ref, query.quote({data: {m: 9}})))
-    assert.deepEqual(got.data, {n: 0, m: 9})
-  })
-
-  it('replace', async () => {
-    const ref = (await create()).ref
-    const got = await client.query(query.replace(ref, query.quote({data: {m: 9}})))
-    assert.deepEqual(got.data, {m: 9})
-  })
-
-  it('delete', async () => {
-    const ref = (await create()).ref
-    await client.query(query.delete_expr(ref))
-    await assertQuery(query.exists(ref), false)
-  })
-
-  it('insert', async () => {
-    const instance = await createThimble({weight: 1})
-    const ref = instance.ref, ts = instance.ts
-    const prevTs = ts - 1
-
-    // Add previous event
-    const inserted = query.quote({data: {weight: 0}})
-    await client.query(query.insert(ref, prevTs, 'create', inserted))
-
-    // Get version from previous event
-    const old = await client.query(query.get(ref, prevTs))
-    assert.deepEqual(old.data, {weight: 0})
-  })
-
-  it('remove', async () => {
-    const instance = await createThimble({weight: 0})
-    const ref = instance.ref
-
-    // Change it
-    const newInstance = await client.query(query.replace(ref, query.quote({data: {weight: 1}})))
-    await assertQuery(query.get(ref), newInstance)
-
-    // Delete that event
-    await client.query(query.remove(ref, newInstance.ts, 'create'))
-
-    // Assert that it was undone
-    await assertQuery(query.get(ref), instance)
-  })
+  it('create', function () {
+    return create().then(function (instance) {
+      assert('ref' in instance);
+      assert('ts' in instance);
+      assert.deepEqual(instance.class, classRef);
+    });
+  });
+
+  it('update', function () {
+    return create().then(function (i) {
+      var ref = i.ref;
+      return client.query(query.update(ref, query.quote({ data: { m: 9 } })));
```

Comment:
Would be nice to call update with a `null` field just to make sure that we can remove a field from an existing instance.